packages/experience-builder-sdk/src/blocks/editor/VisualEditorContext.tsx

```diff
@@ -337,7 +352,7 @@ export function VisualEditorContextProvider({
     return () => {
       window.removeEventListener('message', onMessage);
     };
-  }, [fetchMissingEntities, entityStore, mode]);
+  }, [fetchMissingEntities, entityStore, mode, dataSource]);
```

Comment:
This might be dangerous to set the dataSource as dependency to the `useEffect` in which we `setDataSource` in multiple places. 
This could lead to infinite loops as we had quite recently

Reply:
Not writing out all dependencies in effects can also pose risks as the effect will eventually not pick up the latest value of a state variable.
Also, note that `fetchMissingEntities` depends on `dataSource`. So if `dataSource` changes, it will already update the callback and thus call again the effect. So practically, this shouldn't cause any difference in the actual execution.
Additionally, note in this screenshot that this effect itself is just adding and removing event listeners. So the state is only changed when an event was triggered, not when the effect was called.

![Screenshot 2023-12-27 at 11 46 20](https://github.com/contentful/experience-builder/assets/9327071/adca2fe2-a4b7-421d-8ea2-c1e4e038c0b5)

Reply:
That is true, I was just a bit cautious because something like that was the reason that we had infinite get requests. 
But you are right that the useCallback should have the same effect, so hopefully nothing has changed because of this